src/cmd/package.go

```diff
@@ -664,23 +649,18 @@ func (o *packageInspectImagesOptions) run(cmd *cobra.Command, args []string) err
 		return err
 	}
 
-	// The user may be pulling the package from the cluster or using a built package
-	// since we don't know we don't check this error
-	c, _ := cluster.New(ctx) //nolint:errcheck
+	inspectImageOpts := packager2.InspectPackageImagesOptions{
+		Architecture:            config.GetArch(),
+		Source:                  src,
+		SkipSignatureValidation: o.skipSignatureValidation,
+		PublicKeyPath:           pkgConfig.PkgOpts.PublicKeyPath,
+	}
 
-	pkg, err := packager2.GetPackageFromSourceOrCluster(ctx, c, src, o.skipSignatureValidation, pkgConfig.PkgOpts.PublicKeyPath)
+	result, err := packager2.InspectPackageImages(ctx, inspectImageOpts)
```

Comment:
Not sure if this has been done in 100% of places (though it's worth checking), but we should have required fields like source as direct parameters instead of options